src/workflows-app/SubmissionHistory.js

```diff
@@ -1,9 +1,389 @@
-});
+export const BaseSubmissionHistory = ({ name, namespace, workspace }, _ref) => {
+  // State
+  const [sort, setSort] = useState({ field: 'submission_timestamp', direction: 'desc' });
+  const [pageNumber, setPageNumber] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState(50);
+  const [runSetsData, setRunSetData] = useState();
+  const [runSetsFullyUpdated, setRunSetsFullyUpdated] = useState();
+  const [loading, setLoading] = useState(false);
+
+  const signal = useCancellation();
+  const scheduledRefresh = useRef();
+  const workspaceId = workspace.workspace.workspaceId;
+
+  const loadRunSets = useCallback(
+    async (cbasUrlRoot) => {
+      try {
+        const runSets = await Ajax(signal).Cbas.runSets.get(cbasUrlRoot);
+        const durationEnhancedRunSets = _.map(
+          (r) => _.merge(r, { duration: getDuration(r.state, r.submission_timestamp, r.last_modified_timestamp, isRunSetInTerminalState) }),
+          runSets.run_sets
+        );
+        return _.merge(runSets, { run_sets: durationEnhancedRunSets });
+      } catch (error) {
+        notify('error', 'Error getting run set data', { detail: error instanceof Response ? await error.text() : error });
+      }
+    },
+    [signal]
+  );
+
+  const loadAllRunSets = useCallback(
+    async (cbasProxyUrlDetails) => {
+      try {
+        if (cbasProxyUrlDetails.status !== AppProxyUrlStatus.Ready) {
+          const { cbasProxyUrlState } = await loadAppUrls(workspaceId, 'cbasProxyUrlState');
+
+          if (cbasProxyUrlState.status === AppProxyUrlStatus.Ready) {
+            loadRunSets(cbasProxyUrlState.state).then((runSets) => {
+              if (runSets !== undefined) {
+                setRunSetData(runSets.run_sets);
+                setRunSetsFullyUpdated(runSets.fully_updated);
+              }
+            });
+          } else {
+            const cbasUrlState = cbasProxyUrlState.state;
+            const errorDetails = cbasUrlState instanceof Response ? await cbasUrlState.text() : cbasUrlState;
+            const additionalDetails = errorDetails ? `Error details: ${JSON.stringify(errorDetails)}` : '';
+            notify('warn', 'Error loading Workflows app', {
+              detail: `Workflows app not found. Will retry in 30 seconds. ${additionalDetails}`,
+              timeout: CbasPollInterval - 1000,
+            });
+          }
+        } else {
+          loadRunSets(cbasProxyUrlDetails.state).then((runSets) => {
+            if (runSets !== undefined) {
+              setRunSetData(runSets.run_sets);
+              setRunSetsFullyUpdated(runSets.fully_updated);
+            }
+          });
+        }
+      } catch (error) {
+        notify('error', 'Error getting run set data', { detail: error instanceof Response ? await error.text() : error });
+      }
+    },
+    [workspaceId, loadRunSets]
+  );
+
+  // helper for auto-refresh
+  const refresh = Utils.withBusyState(setLoading, async () => {
```

Comment:
TOL: would it make sense to start this auto-refresh only after CBAS app is in ready state? Otherwise both this and the ` usePollingEffect` on L143 would call Leo to get latest app statuses resulting in duplicate calls?

Reply:
Ah, so only calling it from within `loadWorkflowsApp()` on mounting? I think that makes a lot of sense. We don't want to constantly be calling Leo.

Reply:
No sorry what I meant was only start the auto-refresh process once CBAS proxy url is in ready state. So you could maybe add a check in here to only start the refresh polling after CBAS proxy url has reached Ready state.